File: gameadmin/web/src/api/slots/gamedaystat.ts
/* eslint-disable */
import type { Empty } from "../empty";


export interface SlotsGameDayStatRecord {
  Game?: string;
  Date?: string;
  /** 特殊的Flag就是针对All，代表 */
  Flag?: string;
  EnterCount?: number;
  /** 押注 */
  Flow?: number;
  /** 赢分 */
  Win?: number;
  /** 业绩 */
  YeJi?: number;
  /** 净输赢 */
  NetProfit?: number;
  /** 抽水 */
  Fee?: number;
  /** 个人奖池爆奖励 */
  SelfPoolReward?: number;
}

export interface SlotsGameDayStatDocsReq {
  StartDate?: string;
  EndDate?: string;
  Game?: string;
  Flags?: string[];
}

export interface SlotsGameDayStatDocsResp {
  List?: SlotsGameDayStatRecord[];
}

export class SlotsGameDayStatRpc {
  static async IncBoth(client, req : SlotsGameDayStatRecord) : Promise<[Empty,any]> {
    return await client.send("slots.SlotsGameDayStatRpc/IncBoth", req)
  }
  static async IncAll(client, req : SlotsGameDayStatRecord) : Promise<[Empty,any]> {
    return await client.send("slots.SlotsGameDayStatRpc/IncAll", req)
  }
  static async Docs(client, req : SlotsGameDayStatDocsReq) : Promise<[SlotsGameDayStatDocsResp,any]> {
    return await client.send("slots.SlotsGameDayStatRpc/Docs", req)
  }
}

File: gamecomponents/comm/slotsmongo/gamedaystat.go
package slotsmongo

import (
	"context"
	"fmt"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo/options"
)

type SlotsPlayerGameDayStatRecord struct {
	Game           string
	Pid            int64
	Date           string
	Flag           string //特殊的Flag就是针对All，代表
	Flow           int64  //押注
	Win            int64  //赢分
	YeJi           int64  //业绩
	NetProfit      int64  //净输赢
	Fee            int64  //抽水
	SelfPoolReward int64  //个人奖池爆奖励
}

func IncAll(req *SlotsPlayerGameDayStatRecord) (err error) {
	req.Flag = "All"
	err = IncFlag(req)
	return
}

func IncBoth(req *SlotsPlayerGameDayStatRecord) (err error) {
	err = IncFlag(req)
	if err != nil {
		return
	}

	if req.Flag == "All" {
		return
	}

	req.Flag = "All"
	err = IncFlag(req)
	return
}

func IncFlag(req *SlotsPlayerGameDayStatRecord) (err error) {
	if req == nil {
		return
	}
	if req.Game == "" {
		err = fmt.Errorf("game is empty")
		return
	}
	if req.Pid == 0 {
		err = fmt.Errorf("Pid is zero")
		return
	}
	if req.Date == "" {
		err = fmt.Errorf("date is empty")
		return
	}
	if req.Flag == "" {
		err = fmt.Errorf("flag is empty")
		return
	}

	_, err = SlotsCollection("PlayerGameDayStat").UpdateOne(context.TODO(),
		bson.M{"Game": req.Game, "Pid": req.Pid, "Date": req.Date, "Flag": req.Flag},
		bson.M{"$inc": bson.M{
			"Flow":           req.Flow,
			"Win":            req.Win,
			"YeJi":           req.YeJi,
			"NetProfit":      req.NetProfit,
			"Fee":            req.Fee,
			"SelfPoolReward": req.SelfPoolReward,
		}},
		options.Update().SetUpsert(true))
	return
}
